pkg/model/external_access.go

```diff
@@ -49,25 +62,31 @@ func (b *ExternalAccessModelBuilder) Build(c *fi.ModelBuilderContext) error {
 		glog.V(2).Infof("bastion is in use; won't configure SSH access to master / node instances")
 	} else {
 		for _, sshAccess := range b.Cluster.Spec.SSHAccess {
-			c.AddTask(&awstasks.SecurityGroupRule{
-				Name:          s("ssh-external-to-master-" + sshAccess),
-				Lifecycle:     b.Lifecycle,
-				SecurityGroup: b.LinkToSecurityGroup(kops.InstanceGroupRoleMaster),
-				Protocol:      s("tcp"),
-				FromPort:      i64(22),
-				ToPort:        i64(22),
-				CIDR:          s(sshAccess),
-			})
+			for masterGroupName, masterGroup := range masterGroups {
+				suffix := GetGroupSuffix(masterGroupName, masterGroups)
+				c.AddTask(&awstasks.SecurityGroupRule{
+					Name:          s(fmt.Sprintf("ssh-external-to-master-%s%s", sshAccess, suffix)),
+					Lifecycle:     b.Lifecycle,
+					SecurityGroup: masterGroup,
+					Protocol:      s("tcp"),
+					FromPort:      i64(22),
+					ToPort:        i64(22),
+					CIDR:          s(sshAccess),
+				})
+			}
 
-			c.AddTask(&awstasks.SecurityGroupRule{
-				Name:          s("ssh-external-to-node-" + sshAccess),
-				Lifecycle:     b.Lifecycle,
-				SecurityGroup: b.LinkToSecurityGroup(kops.InstanceGroupRoleNode),
-				Protocol:      s("tcp"),
-				FromPort:      i64(22),
-				ToPort:        i64(22),
-				CIDR:          s(sshAccess),
-			})
+			for nodeGroupName, nodeGroup := range nodeGroups {
+				suffix := GetGroupSuffix(nodeGroupName, nodeGroups)
+				c.AddTask(&awstasks.SecurityGroupRule{
+					Name:          s(fmt.Sprintf("ssh-external-to-node-%s%s", sshAccess, suffix)),
+					Lifecycle:     b.Lifecycle,
+					SecurityGroup: nodeGroup,
+					Protocol:      s("tcp"),
+					FromPort:      i64(22),
+					ToPort:        i64(22),
+					CIDR:          s(sshAccess),
```

Comment:
This is going to be problematic in the shared SG case, if they don't have the same values